locator/load_sketch.hh

```diff
@@ -236,6 +236,19 @@ public:
         }
         return minmax;
     }
+
+    // Returns nullopt if capacity is not known.
+    std::optional<double> get_allocated_utilization(host_id node, const locator::load_stats& stats, uint64_t target_tablet_size) const {
+        if (!_nodes.contains(node)) {
+            return std::nullopt;
+        }
+        auto& n = _nodes.at(node);
+        if (!stats.capacity.contains(node)) {
+            return std::nullopt;
+        }
+        auto capacity = stats.capacity.at(node);
+        return capacity > 0 ? double(n.load() * target_tablet_size) / capacity : 0;
```

Comment:
This implementation is searching node twice each time.
It's a bit less readable but more efficient to use find, like:
```c++
        auto nodes_it = _nodes.find(node);
        if (nodes_it == _nodes.end()) {
            return std::nullopt;
        }
        auto& n = nodes_it.second;
        auto stats_it = stats.capacity.find(node);
        if (stats_it == stats.capacity.end()) {
            return std::nullopt;
        }
        auto capacity = stats_it->second;
        return capacity > 0 ? double(n.load() * target_tablet_size) / capacity : 0;
```

Reply:
Efficiency doesn't matter here, so simplicity wins.